CPU Only backend built fail

Hi, there, I tried to build CPU only backend ("cmake .. -DCMAKE_BUILD_TYPE=Release -DAF_BUILD_CUDA=OFF -DAF_BUILD_OPENCL=OFF
")

When issue "make -j4" I encountered following errors:

/home/michael/fb/arrayfire/src/backend/cpu/blas.cpp:243:31: error: cannot convert ‘const std::complex<float>*’ to ‘cpu::cptr_type<std::complex<float> > {aka const float*}’ in argument passing
                 gemv_func<T>()(CblasColMajor, lOpts, lDims[0], lDims[1], alpha,
                               ^
/home/michael/fb/arrayfire/src/backend/cpu/blas.cpp:247:31: error: cannot convert ‘const std::complex<float>*’ to ‘cpu::cptr_type<std::complex<float> > {aka const float*}’ in argument passing
                 gemm_func<T>()(CblasColMajor, lOpts, rOpts, M, N, K, alpha,
                               ^
/home/michael/fb/arrayfire/src/backend/cpu/blas.cpp: In instantiation of ‘cpu::matmul(const cpu::Array<T>&, const cpu::Array<T>&, af_mat_prop, af_mat_prop)::<lambda(cpu::Param<T>, cpu::CParam<T>, cpu::CParam<T>)> [with T = std::complex<double>]’:
/home/michael/fb/arrayfire/src/backend/cpu/blas.cpp:247:67:   required from ‘struct cpu::matmul(const cpu::Array<T>&, const cpu::Array<T>&, af_mat_prop, af_mat_prop) [with T = std::complex<double>]::<lambda(class cpu::Param<std::complex<double> >, class cpu::CParam<std::complex<double> >, class cpu::CParam<std::complex<double> >)>’
/home/michael/fb/arrayfire/src/backend/cpu/blas.cpp:304:5:   required from ‘cpu::Array<T> cpu::matmul(const cpu::Array<T>&, const cpu::Array<T>&, af_mat_prop, af_mat_prop) [with T = std::complex<double>]’
/home/michael/fb/arrayfire/src/backend/cpu/blas.cpp:344:1:   required from here
/home/michael/fb/arrayfire/src/backend/cpu/blas.cpp:243:31: error: cannot convert ‘const std::complex<double>*’ to ‘cpu::cptr_type<std::complex<double> > {aka const double*}’ in argument passing
                 gemv_func<T>()(CblasColMajor, lOpts, lDims[0], lDims[1], alpha,
                               ^
/home/michael/fb/arrayfire/src/backend/cpu/blas.cpp:247:31: error: cannot convert ‘const std::complex<double>*’ to ‘cpu::cptr_type<std::complex<double> > {aka const double*}’ in argument passing
                 gemm_func<T>()(CblasColMajor, lOpts, rOpts, M, N, K, alpha,
                               ^
In file included from /home/michael/fb/arrayfire/src/backend/cpu/platform.hpp:13:0,
                 from /home/michael/fb/arrayfire/src/backend/cpu/Array.hpp:16,
                 from /home/michael/fb/arrayfire/src/backend/cpu/blas.hpp:10,
                 from /home/michael/fb/arrayfire/src/backend/cpu/blas.cpp:10:
/home/michael/fb/arrayfire/src/backend/cpu/queue.hpp:59:10: error: ‘void cpu::queue::enqueue(F, Args ...) [with F = cpu::matmul(const cpu::Array<T>&, const cpu::Array<T>&, af_mat_prop, af_mat_prop) [with T = std::complex<float>]::<lambda(cpu::Param<std::complex<float> >, cpu::CParam<std::complex<float> >, cpu::CParam<std::complex<float> >)>; Args = {cpu::Array<std::complex<float> >, cpu::Array<std::complex<float> >, cpu::Array<std::complex<float> >}]’, declared using local type ‘cpu::matmul(const cpu::Array<T>&, const cpu::Array<T>&, af_mat_prop, af_mat_prop) [with T = std::complex<float>]::<lambda(cpu::Param<std::complex<float> >, cpu::CParam<std::complex<float> >, cpu::CParam<std::complex<float> >)>’, is used but never defined [-fpermissive]
     void enqueue(const F func, Args... args) {
          ^
/home/michael/fb/arrayfire/src/backend/cpu/queue.hpp:59:10: error: ‘void cpu::queue::enqueue(F, Args ...) [with F = cpu::matmul(const cpu::Array<T>&, const cpu::Array<T>&, af_mat_prop, af_mat_prop) [with T = std::complex<double>]::<lambda(cpu::Param<std::complex<double> >, cpu::CParam<std::complex<double> >, cpu::CParam<std::complex<double> >)>; Args = {cpu::Array<std::complex<double> >, cpu::Array<std::complex<double> >, cpu::Array<std::complex<double> >}]’, declared using local type ‘cpu::matmul(const cpu::Array<T>&, const cpu::Array<T>&, af_mat_prop, af_mat_prop) [with T = std::complex<double>]::<lambda(cpu::Param<std::complex<double> >, cpu::CParam<std::complex<double> >, cpu::CParam<std::complex<double> >)>’, is used but never defined [-fpermissive]

--- a/src/backend/cpu/blas.cpp
+++ b/src/backend/cpu/blas.cpp
@@ -241,12 +241,16 @@ Array<T> matmul(const Array<T> &lhs, const Array<T> &rhs, af_mat_prop optLhs,
                 dim_t incr =
                     (optRhs == AF_MAT_NONE) ? rStrides[0] : rStrides[1];
                 gemv_func<T>()(CblasColMajor, lOpts, lDims[0], lDims[1], alpha,
-                               left.get(), lStrides[1], right.get(), incr, beta,
-                               output.get(), 1);
+                               static_cast<CBT*>(left.get()), lStrides[1],
+                               static_cast<CBT*>(right.get()), incr, beta,
+                               static_cast<BT*>(output.get()), 1);
             } else {
                 gemm_func<T>()(CblasColMajor, lOpts, rOpts, M, N, K, alpha,
-                               left.get(), lStrides[1], right.get(),
-                               rStrides[1], beta, output.get(), output.dims(0));
+                               static_cast<CBT*>(left.get()), lStrides[1],
+                               static_cast<CBT*>(right.get()),
+                               rStrides[1], beta,
+                               static_cast<BT*>(output.get()),
+                               output.dims(0));
             }
         } else {
             int batchSize = oDims[2] * oDims[3];
@@ -269,9 +273,9 @@ Array<T> matmul(const Array<T> &lhs, const Array<T> &rhs, af_mat_prop optLhs,
                 int roff = z * (is_r_d2_batched * rStrides[2]) +
                            w * (is_r_d3_batched * rStrides[3]);
 
-                lptrs[n] = reinterpret_cast<CBT *>(left.get() + loff);
-                rptrs[n] = reinterpret_cast<CBT *>(right.get() + roff);
-                optrs[n] = reinterpret_cast<BT *>(
+                lptrs[n] = static_cast<CBT *>(left.get() + loff);
+                rptrs[n] = static_cast<CBT *>(right.get() + roff);
+                optrs[n] = static_cast<BT *>(
                     output.get() + z * oStrides[2] + w * oStrides[3]);
             }
 

